### PR TITLE
Feature/gh 10363 add quantity to model

### LINKED ISSUE #10363: Add required fields to CPQ models for Quantity and DataType and their mapping in the CPQ normalizer
Add required fields to CPQ models for Quantity and DataType and their mapping in the CPQ normalizer. (Support only integers, no floating point numbers)

Add new fields to model. Quantity should already be available at attribute and value level on configurator model, add it to CPQ model as well.
DataType must be an enumeration on attribute level.

Map new quantity fields from CPQ model to configurator model in normalizer.

### LINKED ISSUE #10376: Take care that for single-valued cstics with qty on value level allowed the correct data is submitted to CPQ.
Single value attributes with value quantity will be shown as if they have attribute quantity. 
In the update case the quantity must be part of the value even though it is shown as attribute quantity.

### PR DESCRIPTION
**Pull request by AlexanderKuschner**
Add quantity and Datatype fields to CPQ models.
Map them in the normalizer.
Ignore DataType 'Quantity on value level' for single value selection attributes

### CODE REVIEW

File: feature-libs/product/configurators/rulebased/core/model/configurator.model.ts [hunk 1]
```diff
@@ -173,6 +173,14 @@ export namespace Configurator {
     GALLERY = 'GALLERY',
   }
 
+  export enum DataType {
+    INPUT_STRING = 'String',
+    INPUT_NUMBER = 'Number',
+    QTY_ATTRIBUTE_LEVEL = 'Quantity',
+    QTY_VALUE_LEVEL = 'Attr.Quantity',
+    N_A = 'N/A',
+    NOT_IMPLEMENTED = 'not_implemented',
```
Comment (Uli-Tiger) [inline feature-libs/product/configurators/rulebased/core/model/configurator.model.ts]: Can't we have for example an INPUT_NUMMBER with QTY_ATTRIBUTE_LEVEL?
in my opinion that would be 2 Enums:
- STRING/NUMBER/NOT_IMPLEMENTED
- QTY_ATTRIBUTE_LEVEL/ QTY_VALUE_LEVE/NO_QUANTITY

File: feature-libs/product/configurators/rulebased/cpq/cpq.models.ts [hunk 2]
```diff
@@ -123,4 +124,12 @@ export namespace Cpq {
     INPUT = 95,
     AUTO_COMPLETE_CUSTOM = 102,
   }
+
+  export enum DataType {
+    INPUT_STRING = 'String',
+    INPUT_NUMBER = 'Number',
+    QTY_ATTRIBUTE_LEVEL = 'Quantity',
+    QTY_VALUE_LEVEL = 'Attr.Quantity',
+    N_A = 'N/A',
```
Comment (Uli-Tiger) [inline feature-libs/product/configurators/rulebased/cpq/cpq.models.ts:133]: this seems to be a duplicate of Configurator.DataType?

**Review by Uli-Tiger — CHANGES_REQUESTED** [review]

File: feature-libs/product/configurators/rulebased/core/model/configurator.model.ts (hunk 1 above)
Comment (AlexanderKuschner) [reply feature-libs/product/configurators/rulebased/core/model/configurator.model.ts]: No. We can not have such combinations

**Comment by Uli-Tiger** [comment]
as discussed on the phone, you could rename the Enum values in the Configurator.models.ts, so it's better understandable and distinct from the cpq.model.

File: feature-libs/product/configurators/rulebased/cpq/cpq.models.ts (hunk 2 above)
Comment (AlexanderKuschner) [reply feature-libs/product/configurators/rulebased/cpq/cpq.models.ts:133]: Enum values in the Configuration.models.ts are renamed

**Review by Uli-Tiger — APPROVED** [review]

**Pull request merged**